lib/translator/common.h

```diff
@@ -130,13 +130,18 @@ bool ReadValue(Span<const uint8_t> data, T& out) {
 }
 
 template <typename T>
-bool WriteValue(const T& data, Span<uint8_t> out) {
+bool WriteValue(const T& data, Span<uint8_t> out, size_t num_bytes) {
```

Comment:
This is going to cause a lot of churn, and additionally undermines safety guarantees as nothing stops `num_bytes` from being larger than `sizeof(T)`. Could it be introduced as an overload for the presumably rare cases that need it, so the rest can use the simpler and shorter function? Or better yet, could the case that needs to write only a prefix of a struct just use a dedicated struct definition containing only that prefix?

Reply:
That's a very valid concern of `num_bytes` exceeding `sizeof(T)`. I think it's worth adding a check to error if we continue with this implementation 

With respect to the overload, this function currently is an overload of the other `WriteValue()` function that used in most cases (https://github.com/googleinterns/scsi2nvme/pull/166/files#diff-e8eb0eb4e8ba5e2a5b4c9411b3673f7dR141), however, I think just having one function where `num_bytes` has a default value would be clearer. 

I think the dedicated struct idea could also be a good option, especially since that right now we are only aware of one such use case. However, if we discover more cases that require writing a prefix of the struct, then having this `WriteValue` function might make it cleaner.

Reply:
Ah, somehow I missed the overload below. Thanks!